sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/V1Beta3.java

```diff
@@ -665,27 +683,67 @@ public Write withProjectId(String projectId) {
       checkNotNull(projectId, "projectId");
       return new Write(projectId);
     }
+  }
+
+  /**
+   * A {@link PTransform} that deletes {@link Entity} objects from Cloud Datastore.
+   *
+   * @see DatastoreIO
+   */
+  public static class Delete extends DatastoreMutation {
+    /**
+     * Note that {@code projectId} is only {@code @Nullable} as a matter of build order, but if
+     * it is {@code null} at instantiation time, an error will be thrown.
+     */
+    public Delete(@Nullable String projectId) {
+      super(projectId, new DeleteFn());
+    }
+
+    /**
+     * Returns a new {@link Delete} that deletes from the Cloud Datastore for the specified project.
+     */
+    public Delete withProjectId(String projectId) {
+      checkNotNull(projectId, "projectId");
+      return new Delete(projectId);
+    }
+  }
+
+  /**
+   * A {@link PTransform} that writes mutations of {@link Entity} objects to Cloud Datastore.
+   *
+   * <p>It requires a {@link DoFn} that tranforms an {@code Entity} object to a {@link Mutation}
+   * <b>Note:</b> Only idempotent Cloud Datastore mutation operations (upsert and delete) should
+   * be used by the {@code DoFn} provided, as the commits are retried when failures occur.
+   */
+  private static class DatastoreMutation extends PTransform<PCollection<Entity>, PDone> {
```

Comment:
Mutate sounds good.

Wrapping would result is boilerplate (overriding each PTranform method and redirecting to Mutate). My idea was to make the Mutate public at some point if users ever needed to provide their own DoFn to this transform directly. But I agree, making abstract is a better idea.